### ISSUE TITLE
Errors in babel-english

### ISSUE THREAD

**Issue by Jazzpirate** [brucemiller/LaTeXML#1505]
When doing `\selectlanguage{english}` in Babel, I get multiple errors of the form:

```
(Processing content /usr/share/texlive/texmf-dist/tex/generic/babel/locale/en/babel-english.tex...
Info:latex:\GenericInfo (babel)             
	at babel-english.tex; line 11 col 0 - line 11 col 1
	Package babel Info: Importing
	data for english
	from babel-en.ini. Reported
UNTIL failed!
Seeking one of T_OTHER[=]
Read so far: T_LETTER[h] T_LETTER[y] T_LETTER[p] T_LETTER[h] T_LETTER[e] T_LETTER[n] T_LETTER[c] T_LETTER[h] T_LETTER[a] T_LETTER[r] T_SPACE[ ] T_CS[\@@] T_CS[\fi] T_CS[\relax] T_CS[\expandafter] T_CS[\iterate] T_CS[\fi] T_CS[\let] T_CS[\iterate] T_CS[\relax] T_CS[\bbl@foreach] T_CS[\bbl@renewlist] T_BEGIN[{] Tokens[\bbl@ifunset,{,b,b,l,@,r,e,n,e,w,@,#,1,},{,},{,\bbl@inisec,[,#,1,],\@@,}] T_END[}] T_CS[\global] T_CS[\let] T_CS[\bbl@renewlist] T_CS[\@empty] T_CS[\def] T_CS[\bbl@elt] T_PARAM[#] T_OTHER[1] T_PARAM[#] T_OTHER[2] T_BEGIN[{] Tokens[\bbl@inireader,#,1,=,#,2,\@@] T_END[}] T_CS[\bbl@cs] T_BEGIN[{] Tokens[r,e,n,e,w,@,\bbl@section] T_END[}] T_CS[\global] T_CS[\bbl@csarg] T_CS[\let] T_BEGIN[{] Tokens[r,e,n,e,w,@,\bbl@section] T_END[}] T_CS[\relax] T_CS[\bbl@cs] T_BEGIN[{] Tokens[s,e,c,p,o,s,t,@,\bbl@section] T_END[}] T_CS[\bbl@csarg] T_BEGIN[{] Tokens[\global,\expandafter,\let] T_END[}] T_BEGIN[{] Tokens[i,n,i,d,a,t,a,@,\languagename] T_END[}] T_CS[\bbl@inidata] T_CS[\bbl@exp] T_BEGIN[{] Tokens[\\,\bbl@add@list,\\,\bbl@ini@loaded,{,\languagename,}] T_END[}] T_CS[\bbl@toglobal] T_CS[\bbl@ini@loaded] T_CS[\fi] T_CS[\bbl@exportkey] T_BEGIN[{] Tokens[p,r,e,h,c] T_END[}] T_BEGIN[{] Tokens[t,y,p,o,g,r,a,p,h,y,.,p,r,e,h,y,p,h,e,n,c,h,a,r] T_END[}] T_BEGIN[{] Tokens[] T_END[}] T_CS[\bbl@exportkey] T_BEGIN[{] Tokens[l,n,b,r,k] T_END[}] T_BEGIN[{] Tokens[t,y,p,o,g,r,a,p,h,y,.,l,i,n,e,b,r,e,a,k,i,n,g] T_END[}] T_BEGIN[{] Tokens[h] T_END[}] T_CS[\bbl@exportkey] T_BEGIN[{] Tokens[l,f,t,h,m] T_END[}] T_BEGIN[{] Tokens[t,y,p,o,g,r,a,p,h,y,.,l,e,f,t,h,y,p,h,e,n,m,i,n] T_END[}] T_BEGIN[{] Tokens[2] T_END[}] T_CS[\bbl@exportkey] T_BEGIN[{] Tokens[r,g,t,h,m] T_END[}] T_BEGIN[{] Tokens[t,y,p,o,g,r,a,p,h,y,.,r,i,g,h,t,h,y,p,h,e,n,m,i,n] T_END[}] T_BEGIN[{] Tokens[3] T_END[}] T_CS[\bbl@exportkey] T_BEGIN[{] Tokens[h,y,p,h,r] T_END[}] T_BEGIN[{] Tokens[t,y,p,o,g,r,a,p,h,y,.,h,y,p,h,e,n,r,u,l,e,s] T_END[}] T_BEGIN[{] Tokens[] T_END[}] T_CS[\bbl@exportkey] T_BEGIN[{] Tokens[h,y,o,t,l] T_END[}] T_BEGIN[{] Tokens[t,y,p,o,g,r,a,p,h,y,.,h,y,p,h,e,n,a,t,e,.,o,t,h,e,r,.,l,o,c,a,l,e] T_END[}] T_BEGIN[{] Tokens[] T_END[}] T_CS[\bbl@exportkey] T_BEGIN[{] Tokens[h,y,o,t,s] T_END[}] T_BEGIN[{] Tokens[t,y,p,o,g,r,a,p,h,y,.,h,y,p,h,e,n,a,t,e,.,o,t,h,e,r,.,s,c,r,i,p,t] T_END[}] T_BEGIN[{] Tokens[] T_END[}] T_CS[\bbl@exportkey] T_BEGIN[{] Tokens[i,n,t,s,p] T_END[}] T_BEGIN[{] Tokens[t,y,p,o,g,r,a,p,h,y,.,i,n,t,r,a,s,p,a,c,e] T_END[}] T_BEGIN[{] Tokens[] T_END[}] T_CS[\bbl@exportkey] T_BEGIN[{] Tokens[c,h,r,n,g] T_END[}] T_BEGIN[{] Tokens[c,h,a,r,a,c,t,e,r,s,.,r,a,n,g,e,s] T_END[}] T_BEGIN[{] Tokens[] T_END[}] T_CS[\bbl@exportkey] T_BEGIN[{] Tokens[d,g,n,a,t] T_END[}] T_BEGIN[{] Tokens[n,u,m,b,e,r,s,.,d,i,g,i,t,s,.,n,a,t,i,v,e] T_END[}] T_BEGIN[{] Tokens[] T_END[}] T_CS[\ifx] T_CS[\bbl@initoload] T_CS[\relax] T_CS[\endinput] T_CS[\fi] T_CS[\endgroup] T_SPACE[U+000a/LF] T_CS[\endinput]

Error:expected:Until:= Missing argument Until:= for Core::Definition::Expandable[\bbl@inipreread Until:=Until:\@@]
	at babel-english.tex; line 14 col 0 - line 14 col 0
	Ended at babel-english.tex; line 14 col 0 - line 14 col 0
	Next token is T_LETTER[h]
	In Core::Definition::Expandable[\bbl@ini... /usr/share/texlive/texmf-dist/tex/generic/babel/babel.def; from line 2678 col 7 to line 2678 col 7
	 <= Core::Gullet[@0x55f733143048] <= Core::Stomach[@0x55f7331512b8] <= Core::Definition::Primitive[Begin] <= ...
```

**Comment by dginev** [comment]
Does the babel test pass upon installation of latexml? You can run it in isolation via:
```
 make test TEST_FILES=t/81_babel.t
```

If that fails, would be nice to also have your texlive version from `pdftex --version`

**Comment by Jazzpirate** [comment]
`All tests successful.` I'm using:

`pdfTeX 3.14159265-2.6-1.40.21 TeX Live 2020/Debian`

**Comment by dginev** [comment]
Minimal example request comes next then, since the `t/babel/csquotes.tex` test contains that command:
```tex
\documentclass{article}

\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
 
\usepackage[french,german,english]{babel}
\usepackage[babel=true]{csquotes}

%\MakeForeignQuote{german}{<}{>}

\begin{document}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Quoting in Current Language}

\selectlanguage{french}
\subsection{\languagename}
\enquote{Citation extérieure}
\enquote{Extérieure \enquote{intérieur} extérieure}

\selectlanguage{german}
\subsection{\languagename}
\enquote{Äußeres Zitat}
\enquote{Äußeres \enquote{inneres} Äußeres}

\selectlanguage{english}
\subsection{\languagename}
\enquote{Outer Quote}
\enquote{Outer \enquote{Inner} Outer}
```

and if that passes, then the issue is more subtle/specific to the exact invocation you're looking at.

**Comment by Jazzpirate** [comment]
The most minimal example already does the job, but looking at that test by comparison, I'm guessing it might be related to me not explicitly loading english as a package option?
```
\documentclass{article}
\usepackage[ngerman]{babel}
\begin{document}
bäh
\selectlanguage{english}
bla
\end{document}
```

**Comment by dginev** [comment]
pdflatex prints the following warning:
```

Package babel Warning: Unknown language `english'. Very likely you
(babel)                requested it in a previous run. Expect some
(babel)                wrong results in this run, which should vanish
(babel)                in the next one. Reported on input line 5.
```

Maybe we can emulate it to a point, and emit a latexml warning in its place.

**Comment by Jazzpirate** [comment]
Wait, mine doesn't print that warning... it also loads the corresponding language file, apparently:
```
This is pdfTeX, Version 3.14159265-2.6-1.40.21 (TeX Live 2020/Debian) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./test.tex
LaTeX2e <2020-02-02> patch level 5
L3 programming layer <2020-07-17>
(/usr/share/texlive/texmf-dist/tex/latex/base/article.cls
Document Class: article 2019/12/20 v1.4l Standard LaTeX document class
(/usr/share/texlive/texmf-dist/tex/latex/base/size10.clo))
(/usr/share/texlive/texmf-dist/tex/generic/babel/babel.sty
(/usr/share/texlive/texmf-dist/tex/generic/babel/babel.def
(/usr/share/texlive/texmf-dist/tex/generic/babel/txtbabel.def))
(/usr/share/texlive/texmf-dist/tex/generic/babel-german/ngerman.ldf
(/usr/share/texlive/texmf-dist/tex/generic/babel-german/ngermanb.ldf)))
(/usr/share/texlive/texmf-dist/tex/latex/l3backend/l3backend-pdfmode.def)
(./test.aux
(/usr/share/texlive/texmf-dist/tex/generic/babel/locale/en/babel-english.tex))
[1{/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}] (./test.aux) )</usr/shar
e/texlive/texmf-dist/fonts/type1/public/amsfonts/cm/cmr10.pfb>
Output written on test.pdf (1 page, 10513 bytes).
Transcript written on test.log.
```

**Comment by dginev** [comment]
Ha, that is funny - it only prints the warning if you first had a run over the file with english loaded, as in `\usepackage[ngerman, english]{babel}`, and then ran without it, as in `\usepackage[ngerman]{babel}`. Indeed if you run without english loaded, `\selectlanguage` seems to autoload `babel-english.tex`.

**Comment by brucemiller** [comment]
Yeah, well, the weird thing is that with english declared,
LaTeXML and LaTeX both load babel-english.ldf, but without
they both load babel-english.tex.  Unfortunately, LaTeXML
chokes on the latter.

**Comment by brucemiller** [comment]
Ugh, some inconsistency in treating EOL here, which causes it to drop the last char...

**Comment by brucemiller** [comment]
This should be fixed by #1513 and #1514. Thanks for the report.
